### ISSUE TITLE
LND will not open channel with c-lightning

### ISSUE THREAD

**Issue by corbimos** [lightningnetwork/lnd#2250]
### Background

I ran into this same problem when I first setup with an earlier version of lnd, after upgrading to 5.0 it went away, but now it's back. 

I had a channel to my c-lightning node but it was forced closed this past week for some reason. When trying to reopen, it tells me the peer is disconnected but this is not the case. The c-lightning node is online and i can make/receive payments with another c-lightning node of mine. 

I am able to connect to another lnd node without an issue. 

```
root@quinsolo:~# bitcoin-lncli.sh connect 036dc6c5fc1a16171f182f48794cf3833654e8018cce0bde540854b19854448f5f@23.100.30.211:9735
{

}
root@quinsolo:~# bitcoin-lncli.sh openchannel 036dc6c5fc1a16171f182f48794cf3833654e8018cce0bde540854b19854448f5f 1000000
[lncli] rpc error: code = Unknown desc = peer disconnected
root@quinsolo:~# bitcoin-lncli.sh connect 0391489a933ab83094d1af43b3df3e8c8d467474649361912f05211d0e68a28fc9@18.235.20.108:9735
{

}
root@quinsolo:~# bitcoin-lncli.sh openchannel 0391489a933ab83094d1af43b3df3e8c8d467474649361912f05211d0e68a28fc9 500000
{
	"funding_txid": "ac757b4b1dc575ffe59d3911b17133b1dce3797208255a9a1e9d7020bc1bab25"
}
```

### Your environment
Running lnd on BTCPay server. The one click deployment on lunanode.

```
root@quinsolo:~# bitcoin-lncli.sh getinfo
{
    "identity_pubkey": "0319fec13dcb6df5b6a0cee49e85b4f6c48e5387d62d67858d6d46615b6e683f81",
    "alias": "0319fec13dcb6df5b6a0",
    "num_pending_channels": 0,
    "num_active_channels": 15,
    "num_peers": 17,
    "block_height": 552080,
    "block_hash": "0000000000000000000c6e7784a0552a4164b62e2378a3bd78e04c1fb6df1af3",
    "synced_to_chain": true,
    "testnet": false,
    "chains": [
        "bitcoin"
    ],
    "uris": [
        "0319fec13dcb6df5b6a0cee49e85b4f6c48e5387d62d67858d6d46615b6e683f81@170.75.175.204:9735"
    ],
    "best_header_timestamp": "1543619574",
    "version": "0.5.0-beta commit=c468a55f9b9ea3483beb970e64d27911af83f911"
}

```
### Expected Behavior 

As you can see above, I was intending to re-open a channel with my c-lightning node, but it shows "peer disconnected." I then connect to the CASA lnd node without issue. 

Logs are attached, i gathered them immediately after the successful opening of the CASA channel.   

[lnd.log](https://github.com/lightningnetwork/lnd/files/2635332/lnd.log)

**Comment by wpaulino** [comment]
From looking at the logs, the connection is being closed after a couple seconds by the c-lightning node. I would check out that node's logs to see why this is happening.

**Comment by corbimos** [comment]
@wpaulino Thanks for the advice. I'll check into the c-lightning logs. 

For what it's worth, i'm having the same problem with both of my c-lightning nodes connecting to my lnd node. So I'm leaning towards it not being isolated to the specific machine.

**Comment by Roasbeef** [comment]
AFAIK, c-lightning won't accept connections with a peer that it has a forced close channel with until it's fully swept, and also 100 blocks have passed since its sweeping transaction.

**Comment by corbimos** [comment]
@Roasbeef Ohhhhh, ok. That would make sense. Thank you for the knowledge. 

But I wonder why it was force closed to begin with? Both servers have been consistently online. 

Also, I am unable to connect to either of my c-lightning nodes from lnd, and one of them did not have a connection previously. 
However, I was able to successfully open a channel with the blockstream store (02df5ffe895c778e10f7742a6c5b8a0cefbe9465df58b92fadeb883752c8107c8f@35.232.170.67), which I believe is c-lightning, so it seems i have a problem with my servers specifically. 

I'll keep tabs on this and maybe make an issue on the c-lightning repo.

Thank you all for the assistance! And thanks for the hard work on this project!

**Comment by corbimos** [comment]
Today i was able to connect to my lnd node from c-lightning for the first time since the original channel was closed, i'm not entirely sure what has changed. 

I was able to open a channel with funds on the c-lightning side,  I will just create an invoice and push them to the lnd side to achieve my goal.

Again, i'm not understanding why it forced closed to begin with, but things seem to jiving now... my least favorite way to get a solution. 

Closing the issue for now, will update if things change.
